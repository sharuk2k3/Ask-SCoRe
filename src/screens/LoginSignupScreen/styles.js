import {StyleSheet} from "react-native";
import {
  COLOR_PRIMARY,
  COLOR_GRAY_SECONDARY,
  COLOR_GRAY_PRIMARY,
} from "../../config/styles";

export default StyleSheet.create({
  WholeTabBar: {
    width: 300,
    height: 40,
    backgroundColor: COLOR_GRAY_PRIMARY,
    alignSelf: "center",
    borderRadius: 20,
    flexDirection: "row",
  },

  LoginTabBar: {
    height: "100%",
    width: "50%",
    backgroundColor: COLOR_GRAY_PRIMARY,
    borderRadius: 20,
  },

  SignupTabBar: {
    height: "100%",
    width: "50%",
    backgroundColor: COLOR_PRIMARY,
    borderRadius: 20,
  },

  ImageView: {
    height: 80,
    width: 80,
    marginBottom: 20,
  },

  BarStyle: {
    textAlign: "center",
    marginTop: 8,
    color: COLOR_GRAY_SECONDARY,
    fontSize: 20,
    fontWeight: "800",
  },

  BarStyle1: {
    textAlign: "center",
    marginTop: 8,
    color: COLOR_PRIMARY,
    fontSize: 20,
    fontWeight: "800",
  },

  ViewStyle: {
    backgroundColor: "white",
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginBottom: "20%",
  },
  ViewStyle1: {backgroundColor: "white", flex: 1},
});
