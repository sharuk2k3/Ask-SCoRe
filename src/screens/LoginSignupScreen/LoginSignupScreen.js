import React, {Component} from "react";
import {View, Text, TouchableOpacity, Image} from "react-native";
import LoginForm from "../../components/LoginForm/LoginForm";
import SignupForm from "../../components/SignupForm/SignupForm";
import styles from "./styles";
import {LOGO} from "../../config/styles.js";

class LoginSignupScreen extends Component {
  state = {showingup: true};
  renderContent() {
    switch (this.state.showingup) {
      case true:
        return (
          <View style={{width: "76%"}}>
            <LoginForm />
          </View>
        );
      case false:
        return (
          <View style={{width: "76%"}}>
            <SignupForm />
          </View>
        );
      default:
        return <Text>Error</Text>;
    }
  }

  render() {
    return (
      <View style={styles.ViewStyle1}>
        <View style={styles.ViewStyle}>
          <View>
            <Image source={LOGO} style={styles.ImageView} />
          </View>
          <View style={styles.WholeTabBar}>
            {this.state.showingup ? (
              <TouchableOpacity
                style={styles.SignupTabBar}
                onPress={() => {
                  this.setState({showingup: true});
                }}>
                <View>
                  <Text style={styles.BarStyle}>LOGIN</Text>
                </View>
              </TouchableOpacity>
            ) : (
              <TouchableOpacity
                style={styles.LoginTabBar}
                onPress={() => {
                  this.setState({showingup: true});
                }}>
                <View>
                  <Text style={styles.BarStyle1}>LOGIN</Text>
                </View>
              </TouchableOpacity>
            )}

            {this.state.showingup === false ? (
              <TouchableOpacity
                style={styles.SignupTabBar}
                onPress={() => {
                  this.setState({showingup: false});
                }}>
                <View>
                  <Text style={styles.BarStyle}>SIGNUP</Text>
                </View>
              </TouchableOpacity>
            ) : (
              <TouchableOpacity
                style={styles.LoginTabBar}
                onPress={() => {
                  this.setState({showingup: false});
                }}>
                <View>
                  <Text style={styles.BarStyle1}>SIGNUP</Text>
                </View>
              </TouchableOpacity>
            )}
          </View>
          {this.renderContent()}
        </View>
      </View>
    );
  }
}
export default LoginSignupScreen;
