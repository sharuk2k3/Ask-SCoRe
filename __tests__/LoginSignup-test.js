import React from "react";
import LoginSignupScreen from "../src/screens/LoginSignupScreen/LoginSignupScreen";
import renderer from "react-test-renderer";

jest.useFakeTimers();

test("renders Question Screen correctly", () => {
  const tree = renderer.create(<LoginSignupScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
