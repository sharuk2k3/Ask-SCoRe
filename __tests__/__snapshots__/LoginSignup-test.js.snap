// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Question Screen correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": "20%",
      }
    }
  >
    <View>
      <Image
        source="test-file-stub"
        style={
          Object {
            "height": 80,
            "marginBottom": 20,
            "width": 80,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#dbdbdb",
          "borderRadius": 20,
          "flexDirection": "row",
          "height": 40,
          "width": 300,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#51AD28",
            "borderRadius": 20,
            "height": "100%",
            "opacity": 1,
            "width": "50%",
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#d4d4d4",
                "fontSize": 20,
                "fontWeight": "800",
                "marginTop": 8,
                "textAlign": "center",
              }
            }
          >
            LOGIN
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#dbdbdb",
            "borderRadius": 20,
            "height": "100%",
            "opacity": 1,
            "width": "50%",
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#51AD28",
                "fontSize": 20,
                "fontWeight": "800",
                "marginTop": 8,
                "textAlign": "center",
              }
            }
          >
            SIGNUP
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "76%",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#dbdbdb",
              "borderRadius": 20,
              "flexDirection": "column",
              "height": 270,
              "marginBottom": "15%",
              "marginTop": "10%",
              "width": "100%",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "white",
                  "borderRadius": 20,
                  "flexDirection": "row",
                  "height": 40,
                  "marginTop": "20%",
                  "paddingLeft": 10,
                  "width": "84%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "gray",
                      "marginRight": 10,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                autoCorrect={false}
                onChangeText={[Function]}
                placeholder="Username"
                placeholderTextColor="gray"
                rejectResponderTermination={true}
                style={
                  Object {
                    "color": "black",
                    "flex": 1,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "white",
                  "borderRadius": 20,
                  "flexDirection": "row",
                  "height": 40,
                  "marginTop": 30,
                  "paddingLeft": 10,
                  "width": "84%",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "alignSelf": "center",
                      "color": "gray",
                      "marginRight": 10,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                autoCorrect={false}
                onChangeText={[Function]}
                placeholder="Password"
                placeholderTextColor="gray"
                rejectResponderTermination={true}
                secureTextEntry={true}
                style={
                  Object {
                    "color": "black",
                    "flex": 1,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "flex-end",
                  "marginRight": "8%",
                  "marginTop": 25,
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "gray",
                  }
                }
              >
                Forgot Password?
              </Text>
            </View>
            <Modal
              animationType="fade"
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              transparent={true}
              visible={false}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "column",
                      "height": "100%",
                      "justifyContent": "center",
                      "width": "100%",
                    },
                    Object {},
                  ]
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "column",
                      "height": "100%",
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#E3E6E7",
                          "borderRadius": 10,
                          "marginLeft": 30,
                          "marginRight": 30,
                          "minWidth": 300,
                        },
                        Object {},
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "padding": 10,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "fontSize": 20,
                            "fontWeight": "bold",
                            "marginBottom": 5,
                            "marginTop": 10,
                            "textAlign": "center",
                          }
                        }
                      >
                        Enter Email
                      </Text>
                      <Text
                        style={
                          Array [
                            Object {
                              "fontSize": 16,
                              "marginBottom": 10,
                              "textAlign": "center",
                            },
                          ]
                        }
                      >
                        Enter email address to send password reset link
                      </Text>
                      <TextInput
                        allowFontScaling={true}
                        autoCapitalize="none"
                        autoCorrect={true}
                        autoFocus={true}
                        clearButtonMode="never"
                        clearTextOnFocus={false}
                        keyboardType="default"
                        maxLength={null}
                        onChangeText={[Function]}
                        onKeyPress={[Function]}
                        placeholder="email"
                        rejectResponderTermination={true}
                        secureTextEntry={false}
                        style={
                          Object {
                            "backgroundColor": "white",
                            "borderColor": "#B0B0B0",
                            "borderRadius": 5,
                            "borderWidth": 1,
                            "color": "rgba(0,0,0,0.54)",
                            "fontSize": 16,
                            "marginBottom": 15,
                            "marginTop": 10,
                            "paddingBottom": 5,
                            "paddingLeft": 10,
                            "paddingTop": 5,
                            "textAlign": "left",
                          }
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </View>
                    <View
                      style={
                        Object {
                          "borderColor": "#B0B0B0",
                          "borderTopWidth": 1,
                          "flex": 1,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "maxHeight": 48,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "flex": 1,
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "borderColor": "#B0B0B0",
                              "borderRightWidth": 5,
                              "color": "#408AE2",
                              "fontSize": 18,
                              "height": 48,
                              "maxHeight": 48,
                              "padding": 10,
                              "textAlign": "center",
                            }
                          }
                        >
                          Cancel
                        </Text>
                      </View>
                      <View
                        style={
                          Object {
                            "backgroundColor": "#B0B0B0",
                            "width": 1,
                          }
                        }
                      />
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "flex": 1,
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#408AE2",
                              "fontSize": 18,
                              "padding": 10,
                              "textAlign": "center",
                            }
                          }
                        >
                          Submit
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </Modal>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 100,
                "left": 15,
                "marginTop": "15%",
                "position": "absolute",
                "top": 200,
                "width": 270,
                "zIndex": 2,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#51AD28",
                  "borderRadius": 25,
                  "height": "45%",
                  "justifyContent": "flex-end",
                  "opacity": 1,
                  "width": "70%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#d4d4d4",
                    "flex": 1,
                    "fontSize": 20,
                    "fontWeight": "800",
                    "marginTop": 12,
                  }
                }
              >
                LOGIN
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
